backend/persistent/datastores/dataworld.js

```diff
@@ -88,6 +88,11 @@ export function query(queryString, connection) {
     body: params
   })
   .then(res => {
+    if (res.status !== 200) {
+      return res.text().then(body => {
```

Comment:
Unfortunately not all error messages are returned as text, some come in the form of an error object.

For example, `401`s produce the following response:

```
{ "code": 401, "request": "06d583b7-afbf-48ca-b1b8-961ed0b092a0", "message": "Unauthorized" }
```

Displaying the entire object is unnecessary since only the content of `message` is useful to the user.

We could replace this bit of code with:
 
```
if (res.status !== 200) {
  return res.text().then(body => {
    try {
      const errorObject = JSON.parse(body);
      throw new Error(errorObject.message);
    } catch (err) {
      if (err.name === 'SyntaxError') {
        throw new Error(body);
      }
      
      throw new Error(err.message);
    }
  });
}
```

Reply:
I've handled this in [Preview.react.js](https://github.com/n-riesco/plotly-database-connector/blob/18b120e3d759888e040f5e5bf4b292c3f4c4e31d/app/components/Settings/Preview/Preview.react.js#L71-L80) so that other connectors benefit from this PR.

Reply:
Agreed, the changes in `Preview.react.js` allows all connectors to display a custom error message instead of simply "stringifying" the error thrown.

The change I'm suggesting has to do with the custom error that the data.world connector throws.

When the response from the API is a string it will be displayed as is, e.g.

When it is a `400` the response from the API is a string specifying the specific problem (e.g. `'(' expected but 'BADSELECT' found... `)

But when the response is an object the entire object is shown, e.g.

When it is a `401` the response from the API is an object that is overly verbose (`{ "code": 401, "request": "06d583b7-afbf-48ca-b1b8-961ed0b092a0", "message": "Unauthorized" }`)

As it currently stands this entire object will be displayed to the user, the change I'm proposing will result in only the contents of the `message` attribute being displayed to the user i.e. `Unauthorized`

Reply:
@kndungu My bad! Your first explanation was clear enough.

I've fixed this in https://github.com/plotly/falcon-sql-client/pull/394/commits/11950d1808af4b8449b02ea43548fcc0a52fa625 . I've also tested it manually.

Please, let me know if it looks OK to you.

Reply:
@n-riesco no worries!

Yes it does! More elegant than my suggested fix :shipit: